pkg/kn/commands/service_get.go

```diff
@@ -0,0 +1,105 @@
+// Copyright © 2018 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"fmt"
+	"strings"
+	"text/tabwriter"
+
+	util "github.com/knative/client/pkg/util"
+	printers "github.com/knative/client/pkg/util/printers"
+	servingv1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+// NewServiceGetCommand represents the get command
+func NewServiceGetCommand(p *KnParams) *cobra.Command {
+
+	serviceGetPrintFlags := genericclioptions.NewPrintFlags("")
+
+	ServiceGetCommand := &cobra.Command{
+		Use:   "get",
+		Short: "Get available services.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := p.ServingFactory()
+			if err != nil {
+				return err
+			}
+			namespace, err := GetNamespace(cmd)
+			if err != nil {
+				return err
+			}
+			service, err := client.Services(namespace).List(v1.ListOptions{})
+			if err != nil {
+				return err
+			}
+			// if output format flag is set, delegate the printing to cli-runtime
+			if cmd.Flag("output").Changed {
+				printer, err := serviceGetPrintFlags.ToPrinter()
+				if err != nil {
+					return err
+				}
+				service.GetObjectKind().SetGroupVersionKind(schema.GroupVersionKind{
+					Group:   "knative.dev",
+					Version: "v1alpha1",
+					Kind:    "Service"})
+				err = printer.PrintObj(service, cmd.OutOrStdout())
+				if err != nil {
+					return err
+				}
+				return nil
+			}
+			// if no output format flag is set, lets print the human readable output
+			printer := printers.NewTabWriter(cmd.OutOrStdout())
```

Comment:
This needs to be a `printer`, implementing the interface here: https://github.com/kubernetes/cli-runtime/blob/master/pkg/printers/interface.go#L34

Reply:
This is a solved problem for kubectl and the human-friendly printer implementation is in place already  https://github.com/kubernetes/kubernetes/blob/master/pkg/printers/tableprinter.go#L49 ,
however its not importable (or we don't want k8s dep) as genericclioptions.
A bunch of utilities can be re-used though from `pkg/printers` (re-use = borrow the code?), and we'll just need to add the column-definitions and their printing logic as done here
https://github.com/kubernetes/kubernetes/blob/master/pkg/printers/internalversion/printers.go#L79  for resources operable via kn.

WDYT? 
Cc: @bbrowning @rhuss

Reply:
@navidshaikh If there are no efforts underway to port that part to cli-runtime anyway, I would try to move that tableprinter stuff to `kn` in an isolated way and if this is successful, create a PR for cli-runtime so that we could use it from there eventually. It depends on the size of the code to move around and how easy it is to isolate it, but I think this could be a strategy from which everyone could benefit.

We could do it the other way round (first move it to cli-runtime), but this is (a) doesn't help for `kn` in the short term and (b) if we fail at cli-runtime, the effort is wasted. That's not the case with the first approach where the only danger is, that we would have to maintain the ported code on our own.

Reply:
@rhuss : makes sense, lets move ahead with first approach as you mentioned.